util/file.py

```diff
@@ -339,3 +339,51 @@ def temp_catted_files(input_files, prefix=None, suffix=None):
         yield fn
     finally:
         os.remove(fn)
+
+def string_to_file_name(string_value):
+    replacements_dict = {
+        "\\": "-", # win directory separator 
+        "/": "-", # posix directory separator 
+        "^": "_", # caret
+        "&": "_and_", # background
+        "\"": "", # double quotes
+        r"'": "", # single quotes
+        r":": "_", # colon (problem for ntfs)
+        r" ": "_", # spaces
+        r"|": "-", # shouldn't confuse a vertical bar for a shell pipe
+        r"!": ".", # not a bash operator
+        r";": ".", # not a terminator
+        r"?": "_", # could be mistaken for a wildcard
+        r"*": "_", # could be mistaken for a wildcard
+        r"`": "_", # no subshells
+        r" -": "_-", # could be mistaken for an argument
+        r" --": "_--", # could be mistaken for an argument
+        r">": "]", # no redirect chars
+        r"<": "[", # no redirect chars
+        r"\\x": "_", # hex char
+        r"\\o": "_", # octal char
+        r"\\u": "", # unicode char
```

Comment:
does this trigger on unicode, or just encoded-unicode? More specifically, would normal unicode (e.g. Sénégal) be accepted as a sample name? I don't see a particular reason to disallow it, especially if DNAnexus is using py3 environments.